rust/private/rustc.bzl

```diff
@@ -36,80 +31,101 @@ CrateInfo = provider(
-        "rustc_env": """File: file containing additional environment variables to set for rustc.""",
-        "dep_env": """File: extra build script environment varibles to be set to direct dependencies.""",
-        "link_flags": """File: file containing flags to pass to the linker""",
+        "flags": "File: file containing additional flags to pass to rustc",
+        "out_dir": "File: directory containing the result of a build script",
+        "rustc_env": "File: file containing additional environment variables to set for rustc.",
+        "dep_env": "File: extra build script environment varibles to be set to direct dependencies.",
+        "link_flags": "File: file containing flags to pass to the linker",
     },
 )
 
 AliasableDepInfo = provider(
-    doc = "",
+    doc = "A provider mapping an alias name to a Crate's information.",
     fields = {
         "name": "str",
         "dep": "CrateInfo",
     },
 )
 
 DepInfo = provider(
+    doc = "A provider contianing information about a Crate's dependencies.",
     fields = {
         "direct_crates": "depset[CrateInfo]",
         "transitive_crates": "depset[CrateInfo]",
         "transitive_dylibs": "depset[File]",
         "transitive_staticlibs": "depset[File]",
         "transitive_libs": "List[File]: All transitive dependencies, not filtered by type.",
         "transitive_build_infos": "depset[BuildInfo]",
-        "dep_env": """File: File with environment variables direct dependencies build scripts rely upon.""",
+        "dep_env": "File: File with environment variables direct dependencies build scripts rely upon.",
     },
 )
 
 def _get_rustc_env(ctx, toolchain):
+    """Gathers rustc environment variables
+
+    Args:
+        ctx (ctx): The current target's rule context object
+        toolchain (rust_toolchain): The current target's rust toolchain context
+
+    Returns:
+        dict: Rustc environment variables
+    """
     version = ctx.attr.version if hasattr(ctx.attr, "version") else "0.0.0"
     major, minor, patch = version.split(".", 2)
     if "-" in patch:
         patch, pre = patch.split("-", 1)
     else:
         pre = ""
     return {
-        "CARGO_PKG_VERSION": version,
+        "CARGO_CFG_TARGET_ARCH": toolchain.target_arch,
+        "CARGO_CFG_TARGET_OS": toolchain.os,
+        "CARGO_PKG_AUTHORS": "",
+        "CARGO_PKG_DESCRIPTION": "",
+        "CARGO_PKG_HOMEPAGE": "",
+        "CARGO_PKG_NAME": ctx.label.name,
         "CARGO_PKG_VERSION_MAJOR": major,
         "CARGO_PKG_VERSION_MINOR": minor,
         "CARGO_PKG_VERSION_PATCH": patch,
         "CARGO_PKG_VERSION_PRE": pre,
-        "CARGO_PKG_AUTHORS": "",
-        "CARGO_PKG_NAME": ctx.label.name,
-        "CARGO_PKG_DESCRIPTION": "",
-        "CARGO_PKG_HOMEPAGE": "",
-        "CARGO_CFG_TARGET_OS": toolchain.os,
-        "CARGO_CFG_TARGET_ARCH": toolchain.target_arch,
+        "CARGO_PKG_VERSION": version,
     }
 
 def get_compilation_mode_opts(ctx, toolchain):
+    """[summary]
```

Comment:
this is the rustc opt / debug flags

Reply:
I had a difficult time trying to find documentation for `toolchain.compilation_mode_opts` (and ultimately didn't find any). Can I ask you to suggest something for me to write here since you seem to have more familiarity with it?

Reply:
I don't remember details, probably best to find the pr that added https://github.com/bazelbuild/rules_rust/blob/fcbb17bbd19b79b507b951e660f04950b4b5c5c5/rust/toolchain.bzl#L6-L10

Reply:
iirc this is the mapping from the bazel opt/dbg flag to rustc args

Reply:
Ok, I updated it and think it's sufficient now.